python/ray/dataframe/dataframe.py

```diff
@@ -2385,31 +2383,24 @@ def itertuples(self, index=True, name='Pandas'):
             name (string, default "Pandas"): The name of the returned
             namedtuples or None to return regular tuples.
         Note:
-            Generators can't be pickeled so from the remote function
+            Generators can't be pickled so from the remote function
             we expand the generator into a list before getting it.
             This is not that ideal.
 
         Returns:
             A tuple representing row data. See args for varying tuples.
         """
-        iters = ray.get([
-            _deploy_func.remote(
-                lambda df: list(df.itertuples(index=index, name=name)),
-                part) for part in self._row_partitions])
-        iters = itertools.chain.from_iterable(iters)
-
-        def _replace_index(row_tuple, idx):
-            # We need to use try-except here because
-            # isinstance(row_tuple, namedtuple) won't work.
-            try:
-                row_tuple = row_tuple._replace(Index=idx)
-            except AttributeError:  # Tuple not namedtuple
-                row_tuple = (idx,) + row_tuple[1:]
-            return row_tuple
+        def itertuples_helper(part, i):
```

Comment:
I was thinking something along the lines of this to resolve the `index` comment below:

```python
index_iter = (obj.index for obj in self._row_metadata.partition_series)

def itertuples_helper(part):
    df = ray.get(part)
    df.columns = self.columns
    df.index = next(index_iter)
    return df.itertuples(index=index, name=name)
```

Something like this.

Reply:
I agree, this is much better. I've updated the PR in this way. Thanks!